lib/tools/broccoli/node-modules-dest-copy.ts

```diff
@@ -75,6 +75,12 @@ export class DestCopy implements IBroccoliPlugin {
 		if(isPlugin) {
 			this.$pluginsService.prepare(dependency).wait();
 		}
+
+		if (dependency.name === constants.TNS_CORE_MODULES_NAME) {
+			let tnsCoreModulesResourcePath = path.join(this.outputRoot, constants.TNS_CORE_MODULES_NAME);
+			shelljs.cp("-Rf", path.join(tnsCoreModulesResourcePath, "*"), this.outputRoot);
+			this.$fs.deleteDirectory(tnsCoreModulesResourcePath).wait();
```

Comment:
You are right, we have to handle this case properly. Take a look at the next commit where the whole function prepareNodeModules is put in a `try-catch` block. Deleting the `app` folder in the `catch` will guarantee the process being triggered on next `prepare <platform>` command.